src/main.rs

```diff
@@ -139,86 +159,168 @@ fn main() {
         } else {
             contents = read_to_string(path).expect("Reading config file should be ok");
         };
-        ConfigFile::new(&contents)
+        ConfigFile::new(&contents).await
     }
-    fn lock(cfg: &mut ConfigFile, path: PathBuf) {
-        let lockfile = cfg.lock();
+    async fn lock(cfg: &mut ConfigFile, path: PathBuf) {
+        let lockfile = cfg.lock().await;
         if path == Path::new("-") {
             println!("{lockfile}");
         } else {
             write(path, lockfile).expect("Writing lock file should be ok");
         }
     }
+
     match args.action {
         Actions::Update => {
-            let c = &mut get_cfg(args.config_file);
-            update(c, true, args.not_verbose);
-            lock(c, args.lock_file);
+            let c = &mut get_cfg(args.config_file).await;
+            update(c, true, args.verbosity).await;
+            lock(c, args.lock_file).await;
         }
         Actions::Purge => {
-            let c = &mut get_cfg(args.config_file);
-            purge(c, args.not_verbose);
-            lock(c, args.lock_file);
+            let c = &mut get_cfg(args.config_file).await;
+            purge(c, args.verbosity);
+            lock(c, args.lock_file).await;
         }
         Actions::Tree {
             charset,
             prefix,
             print_tarballs,
-        } => print!(
+        } => println!(
             "{}",
             tree(
-                &mut get_cfg(args.config_file), // no locking needed
+                &mut get_cfg(args.config_file).await, // no locking needed
                 charset,
                 prefix,
                 print_tarballs
             )
+            .await
         ),
-        Actions::Init { packages } => init(packages).expect("Initializing cfg should be ok"),
+        Actions::Init { packages } => {
+            let buf = stream::iter(packages.into_iter())
+                .map(|pp| async move {
+                    let name = pp.to_string();
+                    pp.into_package()
+                        .await
+                        .expect(format!("Package {name} could not be parsed").as_str())
+                })
+                .buffer_unordered(4);
+            let packages = buf.collect::<Vec<Package>>().await;
+            init(packages).await.expect("Initializing cfg should be ok");
+        }
     }
 }
 
-fn update(cfg: &mut ConfigFile, modify: bool, not_verbose: bool) {
+async fn update(cfg: &mut ConfigFile, modify: bool, v: Verbosity) {
```

Comment:
thats mainly because theres other functions that use it, but i suppose it would make more sense to call those functions from the function that uses it directly.

Reply:
get_cfg and lock use CLI arguments, and i would have to pass the argument around... I wonder if i should just pass the entire args object...

Reply:
it needs to be like that for tests to work